front/app/modules/commercial/posthog_user_tracking/index.ts

```diff
@@ -0,0 +1,54 @@
+import { combineLatest } from 'rxjs';
+
+import appConfigurationStream from 'api/app_configuration/appConfigurationStream';
+import authUserStream from 'api/me/authUserStream';
+
+import {
+  IDestinationConfig,
+  registerDestination,
+} from 'components/ConsentManager/destinations';
+
+import { initializeFor } from 'utils/analytics';
+import { ModuleConfiguration } from 'utils/moduleUtils';
+
+import { initializePosthog } from '../posthog_integration';
+
+declare module 'components/ConsentManager/destinations' {
+  export interface IDestinationMap {
+    posthog: 'posthog';
+  }
+}
+
+const destinationConfig: IDestinationConfig = {
+  key: 'posthog',
+  category: 'analytics',
+  feature_flag: 'posthog_user_tracking',
+  name: () => 'PostHog',
+};
+
+const POSTHOG_API_KEY = process.env.POSTHOG_API_KEY;
+
+const configuration: ModuleConfiguration = {
```

Comment:
Opted to create a separate module for this as the previous one was getting unmaintainable.